core/src/main/java/ca/odell/glazedlists/swing/AutoCompleteSupport.java

```diff
@@ -405,6 +404,182 @@
     private Action originalAquaSelectNextAction;
     private Action originalAquaSelectPreviousAction;
 
+    
+    
+    /**
+     * This class encapsulates the specifics of tracking user input and
+     * using it to search the ComboBox items according to FilterMode of
+     * STARTS_WITH or CONTAINS. {@link #getTextMatchingStrategy() }
+     * is taken into account.
+     */
+    private abstract class UserInputFilter {
+        protected Matcher<String> filterMatcher = Matchers.trueMatcher();
+        protected String input = "";
+
+        /**
+         * This method typically updates the prefix to be the current
+         * value in the ComboBoxEditor; sets up the filterMatcher.
+         */
+        abstract void updateFilter(String userInput);
+        
+        /**
+         * Does this UserInputFilter match the specified string.
+         * @param itemString string to check for match
+         * @return true if match
+         */
+        boolean matches(String itemString) {
+            return filterMatcher.matches(itemString);
+        }
+
+        /**
+         * Given the current state of this UserInputFilter,
+         * examine the ComboBox editor and determine
+         * the user input to use for filtering.
+         * @return 
+         */
+        abstract String determineInput();
+
+        /**
+         * Indicate/highlight the characters input by the user in the ComboBox editor.
+         */
+        abstract void visualizeUserInputText();
+
+        /**
+         * Track location/offset of user input in current combo text.
+         * {@code findInputInString("")} conventionally resets/clears info.
+         * @param matchString find user input in this string
+         */
+        abstract void findInputInString(String matchString);
+
+        /**
+         * Search matchString to find the index of the current input
+         * using the TextMatchingStrategy in effect.
+         * @param matchString search in this string
+         * @return index into matchString
+         */
+        protected int indexOf(String matchString) {
+            Object strategyFactory = getTextMatchingStrategy();
+            if (strategyFactory instanceof TextSearchStrategy.Factory) {
+                TextSearchStrategy finder = ((TextSearchStrategy.Factory)strategyFactory).create(getFilterMode(), input);
+                finder.setSubtext(input);
+                return finder.indexOf(matchString);
+            }
+            return -1;
+        }
+    }
+
+    /**
+     * For STARTS_WITH
+     */
+    private class PrefixFilter extends UserInputFilter
```

Comment:
Some cosmetic change: please adapt curly braces in this class to the existing code style.